libevmasm/SemanticInformation.cpp

```diff
@@ -284,6 +284,37 @@ SemanticInformation::Effect SemanticInformation::memory(Instruction _instruction
 	}
 }
 
+bool SemanticInformation::readsMemory(Instruction const _instruction)
```

Comment:
Why not split the `SemanticInformation::Write` into `Write` and `ReadWrite`? Then we wouldn't need two helpers?

Reply:
Sure, I can do that. How would you order the effects? None, Read, ReadWrite, Write? or None, Read, Write, ReadWrite? Probably either one should be fine.

Reply:
I also think that adding `ReadWrite` would be better. This also mean that we can combine read and write of multiple statements (or functions) into readwrite.

Reply:
I actually started doing this refactoring to see how many things are affected, and it didn't seem that bad, but didn't push a PR doing so.